base/server/python/pki/server/deployment/__init__.py

```diff
@@ -1351,10 +1353,16 @@ def import_perm_sslserver_cert(self, instance, cert):
 
     def create_cert(self, subsystem, tag, request):
 
-        # get a new cert ID from the server
-        logger.info('Creating cert ID for %s cert', tag)
-        request.systemCert.certID = self.client.createCertID(request)
-        logger.info('- cert ID: %s', request.systemCert.certID)
+        cert_id_generator = subsystem.config.get('dbs.cert.id.generator', 'legacy')
+
+        if cert_id_generator == 'legacy':
```

Comment:
As above.

Reply:
Similarly, the `else` should cover all future cases. I updated the comment to make it more general.